elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -389,21 +387,12 @@ private void supplyFilterQueryParameters(Query query, NamedParamPreparedStatemen
      */
     private SQLQuery toPageTotalSQL(Query query, SQLQuery sql, SQLDialect sqlDialect) {
 
-        SQLReferenceTable queryReferenceTable = new DynamicSQLReferenceTable(referenceTable, query);
-        // TODO: refactor this method
-        String groupByDimensions =
-                query.getAllDimensionProjections()
-                        .stream()
-                        .map(SQLColumnProjection.class::cast)
-                        .map((column) -> column.toSQL(queryReferenceTable))
-                        .collect(Collectors.joining(", "));
-
         SQLQuery innerQuery =  SQLQuery.builder()
-                .projectionClause(groupByDimensions)
+                .projectionClause(sql.getProjectionClause())
```

Comment:
Thinking about this, we ought to remove the metrics if possible or replace them with count or something simple.  Imagine if the metric was a data sketch aggregation.  We don't want to do some expensive operation.